### PR TITLE
Upgradable crop join

### PR DESCRIPTION
**Pull request by gbalabasquer**

### CODE REVIEW

File: src/CropJoin.sol [hunk 1]
```diff
@@ -65,6 +115,11 @@ contract CropJoin {
     event Flee();
```
Comment (hexonaut) [inline src/CropJoin.sol:115]: Can remove `Rely`/`Deny` events.

**Review by hexonaut — APPROVED** [review]
Minor change, but other than that looks good.

File: src/CropJoin.sol (hunk 1 above)
Comment (gbalabasquer) [reply src/CropJoin.sol:113]: Thanks, removed.

File: src/CropJoin.sol [hunk 2]
```diff
@@ -33,19 +86,16 @@ interface ERC20 {
...
 // receives tokens and shares them among holders
-contract CropJoin {
-    // --- Auth ---
+contract CropJoinImp {
+    bytes32 slot0;
     mapping (address => uint256) public wards;
-    function rely(address usr) public auth { wards[usr] = 1; emit Rely(msg.sender); }
-    function deny(address usr) public auth { wards[usr] = 0; emit Deny(msg.sender); }
-    modifier auth { require(wards[msg.sender] == 1, "CropJoin/non-authed"); _; }
+    uint256 public live;
```
Comment (kmbarry1) [inline src/CropJoin.sol:92]: These two can be private now to cut down on bytecode size since the proxy exposes both directly.

**Review by kmbarry1 — APPROVED** [review]
One suggestion but overall LGTM.

File: src/CropJoin.sol (hunk 2 above)
Comment (gbalabasquer) [reply src/CropJoin.sol]: Done.

**Comment by gbalabasquer** [comment]
Proceeding merging this. The fuzzing errors seems to fail the same way with the same counter examples in `master` than this branch.
Other errors reported in the fuzzing do not fail now with the counter example in any of the branches. I think it has to do with the lack of determinism running it to an evolving mainnet fork state.
So basically it doesn't seem this branch is adding any difference on the fuzzing test results.

**Pull request merged**